packages/backend/src/open_payments/auth/middleware.ts

```diff
@@ -37,26 +38,45 @@ export function createTokenIntrospectionMiddleware({
         return
       }
       const authService = await ctx.container.use('authService')
-      const grant = await authService.introspect(token)
-      if (!grant) {
+      const tokenInfo = await authService.introspect(token)
+      if (!tokenInfo) {
         ctx.throw(401, 'Invalid Token')
       }
-      const access = grant.findAccess({
-        type,
-        action,
-        identifier: ctx.paymentPointer.url
-      })
+      // TODO
+      // https://github.com/interledger/rafiki/issues/835
+      const access = tokenInfo.access.find(
+        (access) =>
+          access.type === type &&
+          (!access['identifier'] ||
+            access['identifier'] === ctx.paymentPointer.url) &&
+          access.actions.find((tokenAction) => {
+            if (tokenAction == action) {
```

Comment:
```suggestion
            if (tokenAction === action) {
```
I don't think a triple-equals will cause issues here, it'll be just a string comparison at runtime

same below

Reply:
Oops, I originally had this based on a branch using generated OpenAPI types where it _was_ an issue.
I tripled the access type check but missed the actions.